pkg/noderesourcetopology/filter.go

```diff
@@ -64,11 +64,13 @@ func resMatchNUMANodes(nodes NUMANodeList, resources v1.ResourceList, qos v1.Pod
 		// for each requested resource, calculate which NUMA slots are good fits, and then AND with the aggregated bitmask, IOW unset appropriate bit if we can't align resources, or set it
 		// obvious, bits which are not in the NUMA id's range would be unset
 		resourceBitmask := bm.NewEmptyBitMask()
-		for _, numaNode := range nodes {
+		for _, numaNode := range numaNodes {
 			numaQuantity, ok := numaNode.Resources[resource]
-			// if can't find requested resource on the node - skip (don't set it as available NUMA node)
-			// if unfound resource has 0 quantity probably this numa node can be considered
-			if !ok && quantity.Cmp(zeroQuantity) != 0 {
+			// if the requested resource can't be found on the NUMA node, we still need to check
+			// if the resource can be found at the node itself, because there are resources which are not NUMA aligned
+			// or not supported by the topology exporter - if resource was not found at both checks - skip (don't set it as available NUMA node).
+			// if the un-found resource has 0 quantity probably this numa node can be considered.
+			if !ok && !resourceFoundOnNode(resource, quantity, nodeInfo) && quantity.Cmp(zeroQuantity) != 0 {
```

Comment:
ok, we're skipping every resource which is not on the worker, IOW we don't skip if it's on the worker node and its quantity is enough.